cpp/src/io/avro/avro_gpu.cu

```diff
@@ -190,7 +197,44 @@ avro_decode_row(schemadesc_s const* schema,
           skip += schema[i].count;  // Should always be 1
         }
       } break;
+
+      case type_duration: {
+        // A duration logical type annotates Avro fixed type of size 12, which
+        // stores three little-endian unsigned integers that represent durations
+        // at different granularities of time. The first stores a number in
+        // months, the second stores a number in days, and the third stores a
+        // number in milliseconds.
+        CUDF_UNREACHABLE("avro type 'duration' not yet implemented");
+      } break;
+
+      // N.B. These aren't handled yet, see the discussion on
+      //      https://github.com/rapidsai/cudf/pull/12788.  The decoding logic
+      //      is correct, though, so there's no harm in having them here.
+      case type_timestamp_millis: [[fallthrough]];
+      case type_timestamp_micros: [[fallthrough]];
+      case type_local_timestamp_millis: [[fallthrough]];
+      case type_local_timestamp_micros: [[fallthrough]];
+      case type_time_millis: [[fallthrough]];
+      case type_time_micros: {
+        // N.B. time-millis is stored as a 32-bit int, however, cudf expects an
+        //      int64 for DURATION_MILLISECONDS.  From our perspective, the fact
+        //      that time-millis comes from a 32-bit int is hidden from us by
+        //      way of the zig-zag varint encoding, so we can safely treat them
+        //      both as int64_t.  Everything else is 64-bit in both avro and
+        //      cudf.
+        CUDF_UNREACHABLE("avro time/timestamp types not yet implemented");
+        int64_t v = avro_decode_zigzag_varint(cur, end);
```

Comment:
Why is there code after this unreachable?

Reply:
Same breadcrumb reason per [this comment](https://github.com/rapidsai/cudf/pull/12788#discussion_r1128461868).  Once the necessary glue is added in to support these logical types -- then the `CUDF_UNREACHABLE()` and comment can be removed, leaving only the correct decoding logic.

As an outside contributor, I figured these breadcrumbs would be useful to other outside contributors that are unfamiliar with the code base but have been tasked with getting support for logical type X added.  They are obviously less useful for NVIDIA folks familiar with the code and its current limitations.

Reply:
The reason this compiles is probably because the `assert()` inside of `CUDF_UNREACHABLE` is compiled out in Release mode. I think this may cause a compile error in a Debug build. I think it would be better to comment out the code if you want to leave it in.

Reply:
It definitely compiles in debug mode, I've been doing all my testing on that, including versions built with `-G` for the `avro_gpu.cu` and `reader_impl.cu` files.

I don't mind removing the `CUDF_UNREACHABLE()` and just commenting out the `int64_t v = ...` with a supporting comment along the lines of "uncomment this when you're ready to implement support for these types".  I'll do that this afternoon.

Reply:
Ok, good to hear. I was suggesting keeping `CUDF_UNREACHABLE()` but just commenting out the code after it.

Reply:
Ah, understood, thanks for clarifying.  I've made that change locally, will push shortly.